decode_test.go

```diff
@@ -1158,6 +1159,23 @@ func TestParserError(t *testing.T) {
 	expectedErr := &yaml.ParserError{
 		Message: "could not find expected ':'",
 		Line:    2,
+		Column:  0,
+	}
+	assert.DeepEqual(t, expectedErr, asErr)
+}
+
+func TestParserErrorDecoder(t *testing.T) {
+	var v any
+	data := "value: -"
+	err := yaml.NewDecoder(strings.NewReader(data)).Decode(&v)
+	asErr := new(yaml.ParserError)
+	if !errors.As(err, &asErr) {
```

Comment:
You could use assert.ErrorAs here

That's its purpose

Reply:
Thanks. This is the first time to use `assert` library for me. I'll try it.